File: app/src/main/java/com/cyl/music_hnust/ui/adapter/MyViewPagerAdapter.java
package com.cyl.music_hnust.ui.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.List;

/**
 * Created by Monkey on 2015/6/29.
 */
public class MyViewPagerAdapter extends FragmentStatePagerAdapter {

    private String[] mTitles;
    private List<Fragment> mFragments;

    public MyViewPagerAdapter(FragmentManager fm, String[] mTitles, List<Fragment> mFragments) {
        super(fm);
        this.mTitles = mTitles;
        this.mFragments = mFragments;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitles[position];
    }

    @Override
    public Fragment getItem(int position) {
        return mFragments.get(position);
    }

    @Override
    public int getCount() {
        return mFragments.size();
    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/local/adapter/MyViewPagerAdapter.kt
package com.cyl.musiclake.ui.music.local.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

/**
 * Created by yonglong on 2015/6/29.
 */
class MyViewPagerAdapter(fm: androidx.fragment.app.FragmentManager, private var mFragments: MutableList<androidx.fragment.app.Fragment>) : androidx.fragment.app.FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): androidx.fragment.app.Fragment {
        return mFragments[position]
    }

    override fun getCount(): Int {
        return mFragments.size
    }

    fun setFragments(mFragments: MutableList<androidx.fragment.app.Fragment>) {
        this.mFragments = mFragments
    }
}
